### PR TITLE
Bump rke2 to v1.21.5+rke2r1

### PR DESCRIPTION
**Pull request by guangbochen**
- Bump RKE2 to `v1.21.5+rke2r1` https://github.com/harvester/harvester/issues/991
- Bump Rancher to `v2.6-1-rc5` https://github.com/harvester/harvester/issues/1243

### CODE REVIEW

File: scripts/version-rke2 [hunk 1]
```diff
@@ -1 +1 @@
-RKE2_VERSION="v1.21.4-rc1+rke2r4"
\ No newline at end of file
+RKE2_VERSION="v1.21.5-rc2-rke2r1"
```
Comment (bk201) [inline scripts/version-rke2:1]:
```suggestion
RKE2_VERSION="v1.21.5-rc2+rke2r1"
```

File: package/harvester-os/files/usr/share/rancher/rancherd/config.yaml.d/50-defaults.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 rancherValues:
   rancherImagePullPolicy: IfNotPresent
   rancherImage: rancher/rancher
-  rancherImageTag: v2.6.0
+  rancherImageTag: v2.6.1-rc3
```
Comment (gitlawr) [inline package/harvester-os/files/usr/share/rancher/rancherd/config.yaml.d/50-defaults.yaml:4]: Is RKE2 provisioning working in rc3? I don't think https://github.com/rancher/rancher/pull/34867 is included in rc3

Comment (guangbochen) [reply package/harvester-os/files/usr/share/rancher/rancherd/config.yaml.d/50-defaults.yaml:4]: seems not, we need to bump Rancher to `v2.6-head` tagged version and fleet to the `0.3.7-rc1` respectively

**Comment by guangbochen** [comment]
verified all pods are up and running, but will @bk201 to double-check whether the multi-node scenario works, thanks
```
NAMESPACE                   NAME                                                     READY   STATUS      RESTARTS   AGE
cattle-fleet-local-system   fleet-agent-74d7b58467-wwmcm                             1/1     Running     0          13s
cattle-fleet-system         fleet-controller-57fd6bbf6b-ch2f6                        1/1     Running     0          2m41s
cattle-fleet-system         gitjob-5778966b7c-k8cjz                                  1/1     Running     0          2m41s
cattle-monitoring-system    prometheus-rancher-monitoring-prometheus-0               3/3     Running     1          105s
cattle-monitoring-system    rancher-monitoring-grafana-7f54b7d8bc-khx5p              3/3     Running     0          2m5s
cattle-monitoring-system    rancher-monitoring-kube-state-metrics-744b9448f4-gmdww   1/1     Running     0          2m5s
cattle-monitoring-system    rancher-monitoring-operator-754bcd8cb4-6zbhx             1/1     Running     0          2m4s
cattle-monitoring-system    rancher-monitoring-prometheus-adapter-77568b975-9t5h2    1/1     Running     0          2m4s
cattle-monitoring-system    rancher-monitoring-prometheus-node-exporter-x2tv8        1/1     Running     0          2m5s
cattle-system               helm-operation-b88vm                                     0/2     Completed   0          2m51s
cattle-system               helm-operation-k4wvt                                     0/2     Completed   0          2m17s
cattle-system               helm-operation-kfhgr                                     0/2     Completed   0          2m30s
cattle-system               helm-operation-wl5f2                                     0/2     Completed   0          64s
cattle-system               rancher-7c6fc8fd4f-s9b22                                 1/1     Running     0          4m40s
cattle-system               rancher-webhook-55757d8fd7-t7bmc                         1/1     Running     0          2m12s
cattle-system               system-upgrade-controller-679575c7b4-mbfwp               1/1     Running     0          97s
harvester-system            harvester-67fc9778fb-bnqg4                               1/1     Running     0          4m25s
harvester-system            harvester-load-balancer-59bf75f489-9vkzf                 1/1     Running     0          4m25s
harvester-system            harvester-network-controller-manager-77b596f58d-dqx8j    1/1     Running     0          4m25s
harvester-system            harvester-network-controller-manager-77b596f58d-rnnh2    1/1     Running     0          4m25s
harvester-system            harvester-network-controller-qpdx2                       1/1     Running     1          4m16s
harvester-system            harvester-node-disk-manager-4rgn8                        1/1     Running     0          4m16s
harvester-system            harvester-webhook-5955cd87bb-d6zmm                       1/1     Running     0          4m25s
harvester-system            kube-vip-cloud-provider-0                                1/1     Running     0          4m25s
harvester-system            kube-vip-zmk6m                                           1/1     Running     0          4m16s
harvester-system            virt-api-666db46cf9-dqggm                                1/1     Running     0          2m2s
harvester-system            virt-api-666db46cf9-hhkt8                                1/1     Running     0          2m2s
harvester-system            virt-controller-6b989ccb4c-2d5sd                         1/1     Running     0          86s
harvester-system            virt-controller-6b989ccb4c-rczjv                         1/1     Running     0          86s
harvester-system            virt-handler-kbkvl                                       0/1     Running     0          86s
harvester-system            virt-operator-66c5dccf5b-ckl6z                           1/1     Running     0          4m25s
kube-system                 cloud-controller-manager-gc                              1/1     Running     0          4m54s
kube-system                 etcd-gc                                                  1/1     Running     0          4m44s
kube-system                 helm-install-harvester-h9vxv                             0/1     Completed   0          4m40s
kube-system                 helm-install-rancher-monitoring-crd-5dnh4                0/1     Completed   0          4m40s
kube-system                 helm-install-rancher-monitoring-js6f9                    0/1     Completed   4          4m40s
kube-system                 helm-install-rke2-canal-gdmxk                            0/1     Completed   0          4m40s
kube-system                 helm-install-rke2-coredns-7xjnq                          0/1     Completed   0          4m40s
kube-system                 helm-install-rke2-metrics-server-chw82                   0/1     Completed   0          4m40s
kube-system                 helm-install-rke2-multus-xd5x8                           0/1     Completed   0          4m40s
kube-system                 kube-apiserver-gc                                        1/1     Running     0          4m26s
kube-system                 kube-controller-manager-gc                               1/1     Running     0          4m54s
kube-system                 kube-multus-ds-ghbzf                                     1/1     Running     0          4m32s
kube-system                 kube-proxy-gc                                            1/1     Running     0          4m50s
kube-system                 kube-scheduler-gc                                        1/1     Running     0          4m54s
kube-system                 rke2-canal-kj4sc                                         2/2     Running     0          4m30s
kube-system                 rke2-coredns-rke2-coredns-7bb4f446c-l668w                1/1     Running     0          4m31s
kube-system                 rke2-coredns-rke2-coredns-autoscaler-7c58bd5b6c-rp5pk    1/1     Running     0          4m31s
kube-system                 rke2-metrics-server-5df7d77b5b-n4w4t                     1/1     Running     0          3m52s
kube-system                 snapshot-controller-9f68fdd9-4s7r9                       1/1     Running     0          4m25s
kube-system                 snapshot-controller-9f68fdd9-z5gw2                       1/1     Running     0          4m25s
longhorn-system             csi-attacher-76dd5cb78b-572cj                            1/1     Running     0          2m37s
longhorn-system             csi-attacher-76dd5cb78b-pr2fq                            1/1     Running     0          2m37s
longhorn-system             csi-attacher-76dd5cb78b-q99j5                            1/1     Running     0          2m36s
longhorn-system             csi-provisioner-7bd455f64b-67g8g                         1/1     Running     0          2m36s
longhorn-system             csi-provisioner-7bd455f64b-9b8dc                         1/1     Running     0          2m36s
longhorn-system             csi-provisioner-7bd455f64b-z54s5                         1/1     Running     0          2m36s
longhorn-system             csi-resizer-5f57c65487-ftd9z                             1/1     Running     0          2m36s
longhorn-system             csi-resizer-5f57c65487-lv5h6                             1/1     Running     0          2m36s
longhorn-system             csi-resizer-5f57c65487-n2v86                             1/1     Running     0          2m36s
longhorn-system             csi-snapshotter-749df8dfcf-hmdtl                         1/1     Running     0          2m36s
longhorn-system             csi-snapshotter-749df8dfcf-rtrgv                         1/1     Running     0          2m36s
longhorn-system             csi-snapshotter-749df8dfcf-wb76s                         1/1     Running     0          2m36s
longhorn-system             engine-image-ei-d9402c3f-xk67g                           1/1     Running     0          3m8s
longhorn-system             instance-manager-e-bdd1a716                              1/1     Running     0          3m7s
longhorn-system             instance-manager-r-82e79419                              1/1     Running     0          3m6s
```

**Comment by guangbochen** [comment]
Bump the version to `rancher-rc5` release and rke2 to `v1.21.5+rke2r1` respectively

File: scripts/images/rancher-images.txt [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
-docker.io/rancher/rancher:v2.6.0
-docker.io/rancher/rancher-webhook:v0.2.0
+docker.io/rancher/rancher:v2.6.1-rc5
+docker.io/rancher/rancher-webhook:v0.2.1
 docker.io/rancher/shell:v0.1.10
-docker.io/rancher/system-agent:v0.1.0-suc
-docker.io/rancher/system-upgrade-controller:v0.7.5
+docker.io/rancher/shell:v0.1.8
+docker.io/rancher/system-agent:v0.1.1-rc1-suc
+docker.io/rancher/system-upgrade-controller:v0.8.0-rc.1
```
Comment (bk201) [inline scripts/images/rancher-images.txt:10]:
```suggestion
docker.io/rancher/system-upgrade-controller:v0.7.5
```
we need to temporarily use the old version until https://github.com/rancher/rancher/issues/34964 is resolved.

**Comment by bk201** [comment]
Could you cherry-pick [this commit](https://github.com/bk201/harvester-installer/commit/e992fe4f79ecd19c888ae7298208b66ed8df937b) together? We can switch to the formal tag.

File: scripts/images/rancher-images.txt (hunk 3 above)
Comment (guangbochen) [reply scripts/images/rancher-images.txt:10]: I think we can tolerate the air-gap image for the master build if the OS upgrader needs this newer image. @bk201

**Comment by guangbochen** [comment]
Updated and cherry-picked the commit, thanks.

**Review by gitlawr — APPROVED** [review]
LGTM

**Review by bk201 — APPROVED** [review]
LGTM.
Works in air-gapped env.

**Pull request merged**